### DISCUSSION TITLE
Example for simple page extraction from one PDF file

### DISCUSSION THREAD

**Discussion by haegar33** [qpdf/qpdf#1198] [Q&A]
I am confused by the "help" documentation which does only provide examples for super complex task (extract odd pages from 3 pdfs...) but I am struggling to get a simple page extraction from one input pdf:
All these commands do not work:

```
qpdf input.pdf --range 1-4 out.pdf
qpdf input.pdf --pages 1-4 -- out.pdf
qpdf input.pdf --pages . 1-4 -- out.pdf
```

Can you please make the documentation more clear and understandable? Thanks!

**Comment by jberkenbilt** [comment]
Point taken about the lack of simpler examples. Your third command looks right if your goal is to just extract the first four pages. What about it is not working?

**Reply by haegar33** [reply]
Actually, you are right, 3.version is working. I was distracted by some warnings about "object numbers"
So the dot "." stands for the current input file. What is confusing that in some examples you process 2 input files with one as a kind of primary..?

**Reply by jberkenbilt** [reply]
Right, with `--pages`, there is a primary input. All document-level metadata is taken from the primary input. `.` is the primary input. It's a shorthand introduced to avoid the need to repeat the name of that file.
